### ISSUE TITLE
Can JSON decoder be loose on constraints about float? 

### ISSUE THREAD

**Issue by junaid1460** [elixir-protobuf/protobuf#364]
Most JSON encoders just wrap 1.0 to 1, while protobuf gives good enough code generation, can we cast 1 -> 1.0 for floats and doubles?

The alternate solution is to use FloatValue and DoubleValue, that feels like just more work for no reason

**Comment by whatyouhide** [comment]
@junaid1460 thanks for the report! Do you know what other Protobuf libraries (for other languages) do in these cases?

**Comment by junaid1460** [comment]
We've been using protobuf in dart/flutter its json decoder just automatically handles 1 and spits out 1.0. 

eg: 
```js
// 1
{"value": 1} 

// 2
{"value": 2.3}

```

```proto
message Resp {
    float value;
}
```

when we parse `1` with `Resp` it should just take it as float.

**Comment by whatyouhide** [comment]
Gotcha, what do decoders for Ruby and Python do for example?

**Comment by v0idpwn** [comment]
Closed per #385

**Comment by junaid1460** [comment]
Thanks :)
